### PR TITLE
Test refactor

### PR DESCRIPTION
**Pull request by pszmel**

### CODE REVIEW

**Review by postrational — APPROVED** [review]

File: ngraph/test/backend/quantize_dequantize.in.cpp [hunk 1]
```diff
@@ -158,7 +136,7 @@ NGRAPH_TEST(${BACKEND_NAME}, dequantize_zero_offset)
     auto offset = op::Constant::create(input_type, scale_offset_shape, {0});
     auto dequantize = make_shared<op::Dequantize>(X, scale, offset, output_type, quantization_axes);
     auto f = make_shared<Function>(dequantize, ParameterVector{X});
-
+    #if 0
```
Comment (ggalieroc) [inline ngraph/test/backend/quantize_dequantize.in.cpp]: Please remove this pre-processing instance, we only need the #else part of this code

**Review by blesniewski — COMMENTED** [review]
@pszmel - have you verified in some way that the results of the tests you refactored are the same before and after the refactor?

**Review by tomdol — APPROVED** [review]

**Comment by pszmel** [comment]
> @pszmel - have you verified in some way that the results of the tests you refactored are the same before and after the refactor?

I copied the expected results from pre-refactored test and add them to expected_output in refactored tests. I guess that tests would not pass if I failed with copying.

**Review by blesniewski — APPROVED** [review]

**Review by ggalieroc — APPROVED** [review]

File: ngraph/test/backend/max.in.cpp [hunk 2]
```diff
@@ -120,21 +96,12 @@ NGRAPH_TEST(${BACKEND_NAME}, max_to_scalar)
...
-    EXPECT_TRUE(test::all_close_f((vector<float>{4}), read_vector<float>(result)));
-
-    // For some reason I'm feeling extra paranoid about making sure reduction doesn't clobber the
-    // input tensors, so let's do this too.
-    EXPECT_TRUE(test::all_close_f(
-        (vector<float>{1, 2, 3, 4}), read_vector<float>(a), MIN_FLOAT_TOLERANCE_BITS));
+    auto test_case = test::TestCase<TestEngine>(f);
+    test_case.add_input<float>(a);
+    test_case.add_expected_output<float>(Shape{}, {4});
+    test_case.run();
```
Comment (sdurawa) [inline ngraph/test/backend/max.in.cpp:104]: missing MIN_FLOAT_TOLERANCE_BITS option

File: ngraph/test/backend/max.in.cpp [hunk 3]
```diff
@@ -162,21 +125,12 @@ NGRAPH_TEST(${BACKEND_NAME}, max_matrix_columns)
...
-    EXPECT_TRUE(test::all_close_f((vector<float>{5, 6}), read_vector<float>(result)));
-
-    // For some reason I'm feeling extra paranoid about making sure reduction doesn't clobber the
-    // input tensors, so let's do this too.
-    EXPECT_TRUE(test::all_close_f(
-        (vector<float>{1, 2, 3, 4, 5, 6}), read_vector<float>(a), MIN_FLOAT_TOLERANCE_BITS));
+    auto test_case = test::TestCase<TestEngine>(f);
+    test_case.add_input<float>({a});
+    test_case.add_expected_output<float>(shape_rt, {5, 6});
+    test_case.run();
```
Comment (sdurawa) [inline ngraph/test/backend/max.in.cpp:133]: missing MIN_FLOAT_TOLERANCE_BITS option

File: ngraph/test/backend/max.in.cpp [hunk 4]
```diff
@@ -186,21 +140,12 @@ NGRAPH_TEST(${BACKEND_NAME}, max_matrix_rows)
...
-    EXPECT_TRUE(test::all_close_f((vector<float>{2, 4, 6}), read_vector<float>(result)));
-
-    // For some reason I'm feeling extra paranoid about making sure reduction doesn't clobber the
-    // input tensors, so let's do this too.
-    EXPECT_TRUE(test::all_close_f(
-        (vector<float>{1, 2, 3, 4, 5, 6}), read_vector<float>(a), MIN_FLOAT_TOLERANCE_BITS));
+    auto test_case = test::TestCase<TestEngine>(f);
+    test_case.add_input<float>({a});
+    test_case.add_expected_output<float>(shape_rt, {2, 4, 6});
+    test_case.run();
```
Comment (sdurawa) [inline ngraph/test/backend/max.in.cpp:148]: missing MIN_FLOAT_TOLERANCE_BITS option

File: ngraph/test/backend/max.in.cpp (hunk 2 above)
Comment (ggalieroc) [reply ngraph/test/backend/max.in.cpp:104]: It does not apply in this case, as EXPECT_TRUE macro is applied in the input vector a

**Review by sdurawa — APPROVED** [review]

**Pull request merged**